tests/test_docker_wrapper.py

```diff
@@ -42,17 +44,70 @@ def test_json_input_with_format_succeeds(tmp_path):
         f.write(f'JSON{SEPARATOR}'.encode())
         f.write(json.dumps(INPUT_PARAMETERS).encode())
 
-    run_docker_wrapper(tmp_path, input_file)
+    check_wrapper_echoes_db(tmp_path, input_file)
 
 
-def run_docker_wrapper(tmp_path, input_file):
+def test_pickle_input_with_format_succeeds(tmp_path):
+    input_file = create_pickle_input(tmp_path)
+
+    check_wrapper_echoes_db(tmp_path, input_file)
+
+
+def test_wrapper_serializes_pickle_output(tmp_path):
+    input_parameters = {'method': 'hello_world', 'output_format': PICKLE}
+    input_file = create_pickle_input(tmp_path, input_parameters)
+
+    output_file = run_docker_wrapper(input_file, tmp_path)
+
+    with output_file.open('rb') as f:
+        assert f.read(len(PICKLE) + 1).decode() == f'{PICKLE}.'
+
+        result = pickle.loads(f.read())
+        pd.testing.assert_frame_equal(SAMPLE_DB, result)
```

Comment:
In general for all these tests it takes me quite some effort to understand what is going on. Maybe you could add some docstrings or comments so it takes less effort to figure out what exactly you are testing.

Reply:
I tried to clarify it a bit more with some additional docstrings and comments

Reply:
Check